sdks/java/extensions/sql/build.gradle

```diff
@@ -21,10 +21,17 @@ import groovy.json.JsonOutput
 apply from: project(":").file("build_rules.gradle")
 applyJavaNature()
 apply plugin: 'ca.coglinc.javacc'
+apply plugin: "com.diffplug.gradle.spotless"
```

Comment:
This is already applied in build_rules.gradle.

It would be best if we followed the existing pattern where we have a boolean in the JavaNatureConfiguration as to whether this is enabled or not. This would allow the spotless configuration start applying project wide.

Reply:
The fact that the plugin is imported in build_rules.gradle is unknown to this module. That should be treated as encapsulated. I'm OK with adding it as a central toggle this time but we should start moving towards modules configuring themselves.

Reply:
(opting in to centralized definitions as appropriate, of course, but avoiding kitchen sink methods)

Reply:
We should apply rules consistently throughout the project to simplify explaining to users how to contribute. The more projects that don't follow the same consistent behavior, the harder it is to explain all the nuances of contributing within our contribution guide.

Reply:
Done

Reply:
In this case, it is self-explanatory. The error message indicates how to fix it.

Reply:
And the place we should explain to users how to contribute is after they are already in a friendly conversation with us. But anyhow that's all tangential to the engineering issue of designing a build so that modules can opt in to functionality.